compiler/testData/codegen/box/operatorConventions/assignOperatorOverload.kt

```diff
@@ -46,9 +57,9 @@ fun box(): String {
     if (foo.x.value != "OK.Container.set2") return "Fail: ${foo.x.value}"
 
     // Test assign() on null is not called
-    val nullCheck: NullCheck? = null
-    nullCheck?.x = "Fail"
-    if (nullCheckResult != "OK") return "Fail: $nullCheckResult"
+//    val nullCheck: NullCheck? = null
```

Comment:
Why this code is commented out?

Reply:
This case did not work before. I added a test for unsafe diagnostics to frontend tests. But I have to double check this test if now it's working as intended.

Reply:
It seems to work ok now (the same as `plusAssign`. So if I have:
```
data class NullCheck(val x: NullCheckContainer)
data class NullCheckContainer(var value: String)

operator fun NullCheckContainer?.assign(value: String) {
    result = value
}
operator fun NullCheckContainer?.plusAssign(value: String) {
    result = value
}
```
and I do:
```
val nullCheck: NullCheck? = null
nullCheck?.x = "Fail"
nullCheck?.x += "Fail"
```

Then nothing is called in both cases.